From: Matthias Klose <doko@debian.org>
Date: Thu, 20 Oct 2016 04:34:06 +0000
Subject: combinedlog

Preserve backward-compatibility in the way the client IP of a request
is logged by the twisted.web HTTP server.

Bug: https://twistedmatrix.com/trac/ticket/7730
Bug-Debian https://bugs.debian.org/772629
---
 src/twisted/web/http.py          |  2 +-
 src/twisted/web/test/test_web.py | 16 ++++++++--------
 2 files changed, 9 insertions(+), 9 deletions(-)

--- a/src/twisted/web/http.py
+++ b/src/twisted/web/http.py
@@ -2731,7 +2731,7 @@
     referrer = _escape(request.getHeader(b"referer") or b"-")
     agent = _escape(request.getHeader(b"user-agent") or b"-")
     line = (
-        u'"%(ip)s" - - %(timestamp)s "%(method)s %(uri)s %(protocol)s" '
+        u'%(ip)s - - %(timestamp)s "%(method)s %(uri)s %(protocol)s" '
         u'%(code)d %(length)s "%(referrer)s" "%(agent)s"' % dict(
             ip=_escape(ip),
             timestamp=timestamp,
--- a/src/twisted/web/test/test_web.py
+++ b/src/twisted/web/test/test_web.py
@@ -1530,7 +1530,7 @@
 
         self.assertEqual(
             # Client IP
-            b'"1.2.3.4" '
+            b'1.2.3.4 '
             # Some blanks we never fill in
             b'- - '
             # The current time (circa 1234567890)
@@ -1623,7 +1623,7 @@
 
         line = http.combinedLogFormatter(timestamp, request)
         self.assertEqual(
-            u'"evil x-forwarded-for \\x80" - - [13/Feb/2009:23:31:30 +0000] '
+            u'evil x-forwarded-for \\x80 - - [13/Feb/2009:23:31:30 +0000] '
             u'"POS\\x81 /dummy HTTP/1.0" 123 - "evil \\x83" "evil \\x84"',
             line)
 
@@ -1755,7 +1755,7 @@
         self.site._logDateTime = "[%02d/%3s/%4d:%02d:%02d:%02d +0000]" % (
             25, 'Oct', 2004, 12, 31, 59)
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy HTTP/1.0" 123 - "-" "-"\n')
 
 
@@ -1767,7 +1767,7 @@
             25, 'Oct', 2004, 12, 31, 59)
         self.request.method = b'G"T'
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"G\\"T /dummy HTTP/1.0" 123 - "-" "-"\n')
 
 
@@ -1779,7 +1779,7 @@
             25, 'Oct', 2004, 12, 31, 59)
         self.request.uri = b'/dummy"withquote'
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy\\"withquote HTTP/1.0" 123 - "-" "-"\n')
 
 
@@ -1791,7 +1791,7 @@
             25, 'Oct', 2004, 12, 31, 59)
         self.request.clientproto = b'HT"P/1.0'
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy HT\\"P/1.0" 123 - "-" "-"\n')
 
 
@@ -1806,7 +1806,7 @@
             b'referer',
             b'http://malicious" ".website.invalid')
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy HTTP/1.0" 123 - '
             b'"http://malicious\\" \\".website.invalid" "-"\n')
 
@@ -1821,7 +1821,7 @@
         self.request.requestHeaders.addRawHeader(b'user-agent',
                                                  b'Malicious Web" Evil')
         self.assertLogs(
-            b'"1.2.3.4" - - [25/Oct/2004:12:31:59 +0000] '
+            b'1.2.3.4 - - [25/Oct/2004:12:31:59 +0000] '
             b'"GET /dummy HTTP/1.0" 123 - "-" "Malicious Web\\" Evil"\n')
 
 
